bdk_esplora_example/src/main.rs

```diff
@@ -0,0 +1,158 @@
+mod esplora;
+use crate::esplora::Client;
+use bdk_core::{
+    bitcoin::{secp256k1::Secp256k1, Network},
+    ConfirmationTime,
+};
+use bdk_file_store::KeychainStore;
+use bdk_keychain::{
+    miniscript::{Descriptor, DescriptorPublicKey},
+    KeychainTracker,
+};
+
+use std::io::{self, Write};
+
+const DEFAULT_PARALLEL_REQUESTS: u8 = 5;
+
+use bdk_cli::{handle_commands, Args, Commands, Context, Keychain, Parser, Result};
+
+fn main() -> Result<()> {
+    let args = Args::parse();
+    let secp = Secp256k1::default();
+    let (descriptor, mut keymap) =
+        Descriptor::<DescriptorPublicKey>::parse_descriptor(&secp, &args.descriptor)?;
+
+    let mut keychain_tracker = KeychainTracker::default();
+    keychain_tracker
+        .txout_index
+        .add_keychain(Keychain::External, descriptor);
+
+    let internal = args
+        .change_descriptor
+        .map(|descriptor| Descriptor::<DescriptorPublicKey>::parse_descriptor(&secp, &descriptor))
+        .transpose()?;
+
+    let change_keychain = if let Some((internal_descriptor, internal_keymap)) = internal {
+        keymap.extend(internal_keymap);
+        keychain_tracker
+            .txout_index
+            .add_keychain(Keychain::Internal, internal_descriptor);
+        Keychain::Internal
+    } else {
+        Keychain::External
+    };
```

Comment:
I think it's probably a good idea to do this one. It could also just be done in the next PR since I guess you'll need the same code for RPC or electrum.